src/main/java/com/example/tournament/Service/TournamentService.java

```diff
@@ -0,0 +1,213 @@
+package com.example.tournament.Service;
+
+
+import com.example.tournament.DTO.TournamentDto;
+import com.example.tournament.Entity.Match;
+import com.example.tournament.Entity.Team;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.Scanner;
+
+@Service
+public class TournamentService {
+    private List<Team> teamList = new ArrayList<>();
+    private List<Match> matchList = new ArrayList<>();
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private TournamentDto tournamentDto = new TournamentDto();
+    private FileSaveService fileSaveService;
+    private Scanner sc = new Scanner(System.in);
+    private Random random = new Random();
+    private boolean exit;
+
+    @Autowired
+    public TournamentService(FileSaveService fileSaveService) {
+        this.fileSaveService = fileSaveService;
+    }
+
+    public void runProgram() throws IOException {
+        while (!exit) {
+            printTournamentMenu();
+            int choiceOfAction = getInputAction();
+            performActionByChoice(choiceOfAction);
+        }
+    }
+
+    public List<Team> saveTeamToList(String teamName, String coachOfTeam, String captainOfTeam) {
+        Team team = new Team(teamName, coachOfTeam, captainOfTeam);
+        teamList.add(team);
+        return teamList;
+    }
+
+    public void printTournamentMenu() {
```

Comment:
Let's move this to separate component that is responsible for interaction with UI and leave here only logic of generating tournament